engine/servermaster/jobop/mock_operator.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jobop
+
+import (
+	"sync"
+
+	"github.com/stretchr/testify/mock"
+	"golang.org/x/net/context"
+)
+
+// MockJobOperator implements JobOperator
+type MockJobOperator struct {
```

Comment:
Why not generated mock by `mockgen`?

Reply:
We have two styles to use mock interface, one is using `mockgen`, the other one is inheriting `testify.Mock`. Maybe we should unify to use one style in the future 😂

Reply:
`mockgen` supports mock method implementation on demand, which makes the mock structure more general. In this case. I'm not sure whether `testify.Mock` supports this feature.

See an example in https://github.com/pingcap/tiflow/blob/f6a740339057d4ac1ec047ea8402d64c8bcbee7e/engine/internal/pkg/discovery/agent_test.go#L39.

Reply:
changed to use `gomock`, it is better than `testify.Mock`. ptal @gozssky